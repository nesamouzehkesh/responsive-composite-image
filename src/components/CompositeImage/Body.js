import React, {Component} from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledImage = styled.img`
  width: 100%;
  height: 100%;
`;

function Body({src}) {
  return <StyledImage src={src} />;
}

Body.propTypes = {
  src: PropTypes.string.isRequired,
};

export default Body;
