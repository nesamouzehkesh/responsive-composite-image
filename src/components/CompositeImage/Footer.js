import React, {Component} from 'react';
import styled from 'styled-components';

const StyledFooter = styled.div`
  display: flex;
  align-items: center;
  width:100%;
  height:30%;
  background-color: rgba(0, 0, 0, ${props => props.opacity});
  position:absolute;
  bottom:0px;
  z-index: 1;
`;

const StyledTitle = styled.span`
  font-size: 20px;
  font-family: Helvetica, Arial, sans-serif;
  font-weight: bold;
  color: #fff;
  /* Tablets, Ipads (portrait) */
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 22px;
  }
  /* Laptops, Desktops */
  @media all and (min-width: 1025px) and (max-width: 1280px) {
    font-size: 15px;
  }
  /* Desktops */
  @media only screen and (min-width: 1281px) {
    font-size: 15px;
  }
`;

const StyledLogo = styled.img`
  margin: 10px;
  width: 40px;
  /* Tablets, Ipads (portrait) */
  @media (min-width: 768px) and (max-width: 1024px) {
    width: 58px;
  }
  /* Laptops, Desktops */
  @media all and (min-width: 1025px) and (max-width: 1280px) {
    width: 40px;
  }
  /* Desktops */
  @media only screen and (min-width: 1281px) {
    width: 40px;
  }  
`;

function Footer({logo, title, opacity}) {
  return (
    <StyledFooter opacity={opacity}>
      <StyledLogo src={logo} title={title} />
      <StyledTitle>{title}</StyledTitle>
    </StyledFooter>
  );
}

export default Footer;
