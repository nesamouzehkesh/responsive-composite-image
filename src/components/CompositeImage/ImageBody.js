import React from 'react';

function ImageBody({src}) {
  return (
    <div>
      <img src={src} />
    </div>
  );
}
export default ImageBody;
