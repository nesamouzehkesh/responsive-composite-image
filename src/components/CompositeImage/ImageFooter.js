import React from 'react';

function ImageFooter({logo, title}) {
  return (
    <div>
      <img src={logo} title={title} />
      {logo} {title}
    </div>
  );
}
export default ImageFooter;
