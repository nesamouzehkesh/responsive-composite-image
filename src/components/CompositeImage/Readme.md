### CompositeImage component:

This is the main component which accepts 4 props; The `image` prop is provided to it from the state of the app supplied by a mock API. It has the assets provoided for the project. The other 3 props are: 

* width
* margin
* opacity 

which provide some flexibility to developers to adjust these attributes as they please. 

#### Simple CompositeImage example:

```js
const image = {
  src: 'http://entertainment.ie//images_content/rectangle/620x372/the-braxton-brothers-in-home-and-away-139692316993502601-140.jpg',
  logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/55/Seven_Network_logo.svg/1200px-Seven_Network_logo.svg.png',
  title: 'Home and Away',
  production: '7 Network',
};

<CompositeImage image={image} />
```


#### CompositeImage example with margin:
For example if you like to put several `CompositeImage` components together you can set some margin between them by setting the `margin` prop as shown below: 

```js
const images = [{
  src: 'http://entertainment.ie//images_content/rectangle/620x372/the-braxton-brothers-in-home-and-away-139692316993502601-140.jpg',
  logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/55/Seven_Network_logo.svg/1200px-Seven_Network_logo.svg.png',
  title: 'Home and Away',
  production: '7 Network',
},
{
  src: 'http://entertainment.ie//images_content/rectangle/620x372/the-braxton-brothers-in-home-and-away-139692316993502601-140.jpg',
  logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/55/Seven_Network_logo.svg/1200px-Seven_Network_logo.svg.png',
  title: 'Home and Away',
  production: '7 Network',
},
{
  src: 'http://entertainment.ie//images_content/rectangle/620x372/the-braxton-brothers-in-home-and-away-139692316993502601-140.jpg',
  logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/55/Seven_Network_logo.svg/1200px-Seven_Network_logo.svg.png',
  title: 'Home and Away',
  production: '7 Network',
},
{
  src: 'http://entertainment.ie//images_content/rectangle/620x372/the-braxton-brothers-in-home-and-away-139692316993502601-140.jpg',
  logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/55/Seven_Network_logo.svg/1200px-Seven_Network_logo.svg.png',
  title: 'Home and Away',
  production: '7 Network',
}];

images.map(image => <CompositeImage image={image} margin="5px" />)
```

#### CompositeImage example with margin & opacity:
This is another use case of `CompositeImage` component with both `margin` and `opacity`set to desired values:

```js
const image = {
  src: 'http://entertainment.ie//images_content/rectangle/620x372/the-braxton-brothers-in-home-and-away-139692316993502601-140.jpg',
  logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/55/Seven_Network_logo.svg/1200px-Seven_Network_logo.svg.png',
  title: 'Home and Away',
  production: '7 Network',
};

<div>
  <CompositeImage image={image} margin="10px" opacity="1"/>
  <CompositeImage image={image} margin="10px" opacity="0.4"/>
</div>
```

#### CompositeImage example with different widths:
This is another use case of `CompositeImage` component different `width` values:

```js
const image = {
  src: 'http://entertainment.ie//images_content/rectangle/620x372/the-braxton-brothers-in-home-and-away-139692316993502601-140.jpg',
  logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/5/55/Seven_Network_logo.svg/1200px-Seven_Network_logo.svg.png',
  title: 'Home and Away',
  production: '7 Network',
};

<div>
  <CompositeImage image={image} margin="10px" />
  <CompositeImage image={image} margin="10px" width={{large: "500px", medium: "400px", small: "200px"}}/>
  <CompositeImage image={image} margin="10px" width={{large: "250px", medium: "250px"}}/>
</div>
```