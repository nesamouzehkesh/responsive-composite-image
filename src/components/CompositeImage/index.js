import React, {Component} from 'react';
import styled from 'styled-components';
import Body from './Body';
import Footer from './Footer';

const StyledCompositeImage = styled.div` 
  width:${props => props.width.small || '100%'};
  background-color:#000;
  position:relative;
  display: inline-block;
  margin: ${props => props.margin};
  /* Tablets, Ipads (portrait) */
  @media (min-width: 768px) and (max-width: 1024px) {
    width: ${props => props.width.medium || '450px'};
  }
  /* Laptops, Desktops */
  @media all and (min-width: 1025px) and (max-width: 1280px) {
    width: ${props => props.width.large || '315px'};
  }
  /* Desktops */
  @media only screen and (min-width: 1281px) {
    width: ${props => props.width.large || '315px'};
  }  
`;

function CompositeImage({image, width, margin, opacity}) {
  return (
    <StyledCompositeImage width={width} margin={margin}>
      <Body src={image.src} />
      <Footer title={image.title} logo={image.logo} opacity={opacity} />
    </StyledCompositeImage>
  );
}

CompositeImage.defaultProps = {
  width: {},
  margin: '0px',
  opacity: '0.7',
};

export default CompositeImage;
