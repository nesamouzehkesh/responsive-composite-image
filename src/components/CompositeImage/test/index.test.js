import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CompositeImage from '../';

const mockImage = {
  src: './media/tile.jpg',
  logo: './media/logo.png',
  title: 'Home and Away',
  production: '7 Network',
};

describe ('Test CompositeImage component', () => {
  it ('renders CompositeImage component normally and checks against snapshot', () => {
    const element = renderer
      .create (<CompositeImage image={mockImage} />)
      .toJSON ();

    expect (element).toMatchSnapshot ();
  });

  it ('renders CompositeImage component normally with margin and checks against snapshot', () => {
    const element = renderer
      .create (<CompositeImage image={mockImage} margin="10px" />)
      .toJSON ();

    expect (element).toMatchSnapshot ();
  });

  it ('renders CompositeImage component normally with opacity and checks against snapshot', () => {
    const element = renderer
      .create (<CompositeImage image={mockImage} opacity="1" />)
      .toJSON ();

    expect (element).toMatchSnapshot ();
  });
});
