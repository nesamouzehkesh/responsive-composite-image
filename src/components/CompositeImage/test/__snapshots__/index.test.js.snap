// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CompositeImage component renders CompositeImage component normally and check with snapshot 1`] = `
<div
  className="sc-ifAKCX cLaDgU"
  width={Object {}}
>
  <img
    className="sc-bdVaJa ehJlOl"
    src="./media/tile.jpg"
  />
  <div
    className="sc-bwzfXH iuahvm"
    opacity="0.7"
  >
    <img
      className="sc-bxivhb jvsqbT"
      src="./media/logo.png"
      title="Home and Away"
    />
    <span
      className="sc-htpNat cThCYu"
    >
      Home and Away
    </span>
  </div>
</div>
`;

exports[`Test CompositeImage component renders CompositeImage component normally and checks against snapshot 1`] = `
<div
  className="sc-ifAKCX cLaDgU"
  width={Object {}}
>
  <img
    className="sc-bdVaJa ehJlOl"
    src="./media/tile.jpg"
  />
  <div
    className="sc-bwzfXH iuahvm"
    opacity="0.7"
  >
    <img
      className="sc-bxivhb jvsqbT"
      src="./media/logo.png"
      title="Home and Away"
    />
    <span
      className="sc-htpNat cThCYu"
    >
      Home and Away
    </span>
  </div>
</div>
`;

exports[`Test CompositeImage component renders CompositeImage component normally with margin and check with snapshot 1`] = `
<div
  className="sc-ifAKCX lhRcTC"
  width={Object {}}
>
  <img
    className="sc-bdVaJa ehJlOl"
    src="./media/tile.jpg"
  />
  <div
    className="sc-bwzfXH iuahvm"
    opacity="0.7"
  >
    <img
      className="sc-bxivhb jvsqbT"
      src="./media/logo.png"
      title="Home and Away"
    />
    <span
      className="sc-htpNat cThCYu"
    >
      Home and Away
    </span>
  </div>
</div>
`;

exports[`Test CompositeImage component renders CompositeImage component normally with margin and checks against snapshot 1`] = `
<div
  className="sc-ifAKCX lhRcTC"
  width={Object {}}
>
  <img
    className="sc-bdVaJa ehJlOl"
    src="./media/tile.jpg"
  />
  <div
    className="sc-bwzfXH iuahvm"
    opacity="0.7"
  >
    <img
      className="sc-bxivhb jvsqbT"
      src="./media/logo.png"
      title="Home and Away"
    />
    <span
      className="sc-htpNat cThCYu"
    >
      Home and Away
    </span>
  </div>
</div>
`;

exports[`Test CompositeImage component renders CompositeImage component normally with opacity and check with snapshot 1`] = `
<div
  className="sc-ifAKCX cLaDgU"
  width={Object {}}
>
  <img
    className="sc-bdVaJa ehJlOl"
    src="./media/tile.jpg"
  />
  <div
    className="sc-bwzfXH dpcdFg"
    opacity="1"
  >
    <img
      className="sc-bxivhb jvsqbT"
      src="./media/logo.png"
      title="Home and Away"
    />
    <span
      className="sc-htpNat cThCYu"
    >
      Home and Away
    </span>
  </div>
</div>
`;

exports[`Test CompositeImage component renders CompositeImage component normally with opacity and checks against snapshot 1`] = `
<div
  className="sc-ifAKCX cLaDgU"
  width={Object {}}
>
  <img
    className="sc-bdVaJa ehJlOl"
    src="./media/tile.jpg"
  />
  <div
    className="sc-bwzfXH dpcdFg"
    opacity="1"
  >
    <img
      className="sc-bxivhb jvsqbT"
      src="./media/logo.png"
      title="Home and Away"
    />
    <span
      className="sc-htpNat cThCYu"
    >
      Home and Away
    </span>
  </div>
</div>
`;
